testcases/PowerNVDump.py

```diff
@@ -1324,6 +1330,262 @@ def runTest(self):
+                self.get_watchdog_tool()
+                self.kernel_crash(crash_type="watchdog")
+            if not self.cv_HMC.get_lpar_state() == "Running":
+                self.fail("System state is mismatching after the watchdog event,Please check logs")
+
+    def check_wd_overNFS(self):
+        '''
+        Function to execute watchdog test case and collect
+        crash dump over remote based NFS directory.
+        '''
+        conf = OpTestConfiguration.conf
+        self.dump_server_ip = conf.args.dump_server_ip
+        self.instance_of_KernelCrash_kdumpNFS = KernelCrash_KdumpNFS()
+        self.instance_of_KernelCrash_kdumpNFS.setUp()
+        if self.check_module_support():
+            self.module_unload("pseries_wdt")
+            self.module_load_with_parameters("pseries_wdt", 60, 2)
+
+            if self.validate_timeout_and_action:
+                if not (self.dump_server_ip or self.dump_server_pw):
+                    raise self.skipTest("Provide --dump-server-ip and --dump-server-pw "
+                                        "for network dumps")
+                self.setup_pwdless_auth()
+                self.setup_test("net")
+                self.instance_of_KernelCrash_kdumpNFS.setup_nfs()
+                log.info("=============== Testing kdump/fadump over nfs ===============")
+                self.get_watchdog_tool()
+                boot_type = self.kernel_crash(crash_type="watchdog")
+                self.verify_dump_file(boot_type, dump_place="net")
+                self.setup_test("net")
+
+    def check_wd_overSAN(self):
+        '''
+        Function to execute watchdog test case and collect
+        crash dump over remote based NFS directory.
+        '''
+        self.instance_of_KernelCrash_kdumpSAN = KernelCrash_KdumpSAN()
+        self.instance_of_KernelCrash_kdumpSAN.setUp()
+        if self.check_module_support():
+            self.module_unload("pseries_wdt")
+            self.module_load_with_parameters("pseries_wdt", 60, 2)
+            if self.validate_timeout_and_action:
+                self.instance_of_KernelCrash_kdumpSAN.setup_san()
+                log.info("=============== Testing kdump/fadump over SAN ===============")
+                self.get_watchdog_tool()
+                boot_type = self.kernel_crash(crash_type="watchdog")
+                self.verify_dump_file(boot_type)
+                self.setup_test()
+
+    def check_wd_localdisk(self):
+        '''
+        Function to execute watchdog test case and collect
+        crash dump over remote based NFS directory.
+        '''
+        if self.check_module_support():
+            if self.distro == "rhel":
+                self.cv_HOST.host_check_command("kdumpctl")
+                obj = OpTestInstallUtil.InstallUtil()
+                if not obj.update_kernel_cmdline(self.distro, args="crashkernel=2G-16G:512M,16G-64G:1G,64G-128G:2G,128G-:4G",
+                                                 reboot=True, reboot_cmd=True):
```

Comment:
any reason to have this only for localdisk ?  is this not required for other SAN, NFS cases ?

Reply:
@abdhaleegit yes . 
when running individual scenarios, its essential to add this and added this check list for every test case.